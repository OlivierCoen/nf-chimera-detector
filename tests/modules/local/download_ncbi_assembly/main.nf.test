nextflow_process {

    name "Test Process DOWNLOAD_NCBI_ASSEMBLY"
    script "modules/local/download_ncbi_assembly/main.nf"
    process "DOWNLOAD_NCBI_ASSEMBLY"
    tag "download_ncbi_assembly"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = [
                    [ taxid: "wesh"],
                    "GCF_000870065.1"
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
