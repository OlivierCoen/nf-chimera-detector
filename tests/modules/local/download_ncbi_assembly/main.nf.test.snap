{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "taxid": "wesh"
                        },
                        "GCF_000870065.1_ViralProj17669_genomic.fna:md5,0a3e94ee7ca26b109e27d6952f04879f"
                    ]
                ],
                "1": [
                    [
                        "DOWNLOAD_NCBI_ASSEMBLY",
                        "ncbi-datasets-cli",
                        "18.5.0"
                    ]
                ],
                "assembly": [
                    [
                        {
                            "taxid": "wesh"
                        },
                        "GCF_000870065.1_ViralProj17669_genomic.fna:md5,0a3e94ee7ca26b109e27d6952f04879f"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T10:01:12.804369681"
    }
}